main.go

```diff
@@ -48,143 +49,132 @@ Usage:
 
 	flag.Parse()
 
-	queue := make(chan string)
-
 	if *hostOnly {
 		fmt.Printf("Crawling URL[%s]\n", *link)
 	} else {
 		fmt.Printf("Crawling All Links in URL[%s]\n", *link)
 	}
 
-	// Go routine puts url flag into channel for queuing.
-	go func() {
-		queue <- *link
-	}()
-
-	hostURL, err := url.Parse(*link)
+	path, err := url.Parse(*link)
 	if err != nil {
+		fmt.Printf("Error: Unparsable URI: %s\n", *link)
+		os.Exit(-1)
 		return
 	}
 
+	// deadLinks provides a lists to contain all failed url links that are
+	// considered deadlinks.
+	var deadLinks []*linkReport
+
+	dead := collectFrom(path, !*hostOnly)
+
 	for {
 		select {
-		case uri, ok := <-queue:
+		case link, ok := <-dead:
 			if !ok {
+				fmt.Println("--------------------DEAD LINKS------------------------------")
+
+				if len(deadLinks) > 0 {
+					for _, f := range deadLinks {
+						fmt.Printf(`
+			URL: %s
+			Status Code: %d
+
+			`, f.Link, f.Status)
+					}
+				} else {
+					fmt.Println("No Dead Links Found! Woot!")
+				}
+
+				fmt.Println("------------------------------------------------------------")
+				os.Exit(-1)
 				return
 			}
 
-			queueLinks(hostURL.Host, uri, queue, *hostOnly)
-
-		case <-time.After(1 * time.Minute):
-			fmt.Println("Ending crawler. Goodbye!")
-			fmt.Println("--------------------DEAD LINKS------------------------------")
-
-			for _, f := range deadLinks {
-				fmt.Println(f)
-			}
-
-			fmt.Println("------------------------------------------------------------")
-
-			return
+			deadLinks = append(deadLinks, link)
 		}
 	}
 
 }
 
-// queueLinks is used for making http calls to the queued links in queue channel
+// collectFrom uses a recursive function to map out the needed lists of links to.
+// It returns a channel through which the acceptable links can be crawled from.
+func collectFrom(path *url.URL, doExternals bool) <-chan *linkReport {
```

Comment:
You can make the returning channel a value of linkReport.  

Such as: <-chan linkReport. The link report type only contains primitive values such as string and int.